sql/insert.go

```diff
@@ -199,7 +199,9 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 		if log.V(2) {
 			log.Infof("CPut %s -> NULL", prettyKey(primaryIndexKey, 0))
 		}
-		b.CPut(primaryIndexKey, nil, nil)
+		// This is subtle: []byte(nil) != interface{}(nil). A []byte(nil) creates
+		// an empty value. An interface{}(nil) deletes the value.
+		b.CPut(primaryIndexKey, []byte(nil), nil)
```

Comment:
I wasn't intentionally optimizing. I think old go compilers allocated for `[]byte{}`, but new ones don't.